discord/ext/tasks/__init__.py

```diff
@@ -166,7 +183,27 @@ async def _loop(self, *args: Any, **kwargs: Any) -> None:
                     self._last_iteration = self._next_iteration
                     self._next_iteration = self._get_next_sleep_time()
                 try:
-                    await self.coro(*args, **kwargs)
+                    token = _current_loop_ctx.set(self._current_loop)
+                    if not self.overlap:
+                        await self.coro(*args, **kwargs)
+                    else:
+
+                        async def run_with_semaphore():
+                            async with self._semaphore:
+                                await self.coro(*args, **kwargs)
+
+                        task = asyncio.create_task(
+                            (
+                                self.coro(*args, **kwargs)
+                                if self.overlap is True
+                                else run_with_semaphore()
+                            ),
+                            name=f"pycord-loop-{self.coro.__name__}-{self._current_loop}",
+                        )
+                        task.add_done_callback(lambda t: self._tasks.discard(t))
```

Comment:
This is actually overcomplicating it, you can just do:
```suggestion
                        task.add_done_callback(self._tasks.discard)
```

Reply:
good to know, i was following copilot suggestion here